### ISSUE TITLE
Why shouldn't I epoch your package

### ISSUE THREAD

**Issue by leigh123linux** [negativo17/libglvnd#1]
You have epoch all the rpmfusion nvidia packages so give me a good reason why I shouldn't do the same to you?

Purposed change for the official libglvnd package

```
--- a/libglvnd.spec	2016-12-31 10:00:18.132868787 +0000
+++ b/libglvnd.spec	2016-12-31 09:59:58.497556781 +0000
@@ -2,6 +2,7 @@
 %global shortcommit0 %(c=%{commit0}; echo ${c:0:7})
 
 Name:           libglvnd
+Epoch:          1
 Version:        0.2.999
 Release:        6%{?commit0:.git%{shortcommit0}}%{?dist}
 Summary:        The GL Vendor-Neutral Dispatch library
@@ -29,6 +30,15 @@ BuildRequires:  xorg-x11-server-Xvfb
 %global         __requires_exclude ^(libGL\\.so.*|libEGL\\.so.*|libGLESv2\\.so.*)$
 }
 
+Obsoletes:  libglvnd-egl < %{?epoch}:%{version}-%{release}
+Provides:   libglvnd-egl = %{?epoch}:%{version}-%{release}
+Obsoletes:  libglvnd-gles < %{?epoch}:%{version}-%{release}
+Provides:   libglvnd-gles = %{?epoch}:%{version}-%{release}
+Obsoletes:  libglvnd-glx < %{?epoch}:%{version}-%{release}
+Provides:   libglvnd-glx = %{?epoch}:%{version}-%{release}
+Obsoletes:  libglvnd-opengl < %{?epoch}:%{version}-%{release}
+Provides:   libglvnd-opengl = %{?epoch}:%{version}-%{release}
+
 
 %description
 libglvnd is an implementation of the vendor-neutral dispatch layer for
@@ -36,7 +46,9 @@ arbitrating OpenGL API calls between mul
 
 %package        devel
 Summary:        Development files for %{name}
-Requires:       %{name}%{?_isa} = :%{version}-%{release}
+Requires:       %{name}%{?_isa} = %{?epoch}:%{version}-%{release}
+Obsoletes:      libglvnd-core-devel < %{?epoch}:%{version}-%{release}
+Provides:       libglvnd-core-devel = %{?epoch}:%{version}-%{release}
 
 %description devel
 The %{name}-devel package contains libraries and header files for
```

We can have a epoch war or you can remove them from the nvidia packages and delete your libglvnd package.

**Comment by scaronni** [comment]
Hello,

at the beginning (2013) I started rebasing just the packages I needed the most and wanted to preserve compatibility (something like an overlay to RPMFusion I would say), but things have changed across the years as I'm helping a few friends at universities, so in the end I also rebase a lot of packages that are in base Fedora repositories and (especially) CentOS/RHEL media even across the same release life span, and ship things with dubious license implications. I guess this stuff can't be shipped anywhere properly for license or policy reasons.

I'm not adding every single package out there, i'm adding just what I need/use myself and unless I start utilizing something new myself I will not add it here. Also no build system or external contributors, so I guess this is really not a competitor for RPMFusion, as for the most part I guess it's already not compatible.

When back from holiday I'm also planning to get rid of all the stuff that I keep (like akmods, package replacements, etc.) which I really don't use and just state that the repositories are not compatible and tested with others, as this is no longer feasible for me, so feel free to bump all the epochs of whatever you want, I have absolutely no interest in any epoch war. If you want RPMFusion to obsolete/provide over every other repository that is out there feel free to proceed. Good hunt :)

To the specifics of libglvnd, there was a moment that in Fedora there was a 0.2.x snapshot already and I needed 0.1.x, so I had to bump the epoch; so I guess it's stuck here forever. If it will ever get bumped in Fedora I can actually remove it.

Cheers,
--Simone

**Comment by leigh123linux** [comment]
Hi Simone,

If the fedora libglvnd package works ok with your nvidia packages the above patch would mean you no longer needed to ship a libglvnd package.
Looking at your nvidia packages it should be possible to remove all the rpmfusion obsoletes and provides as there is no longer a naming conflict.
Are both points acceptable for you?

Regards 

Leigh

**Comment by scaronni** [comment]
Hello,

> If the fedora libglvnd package works ok with your nvidia packages the above patch would mean you no longer needed to ship a libglvnd package

Should be fine. I'm on holiday until next week, so no chance for me to test.

> Looking at your nvidia packages it should be possible to remove all the rpmfusion obsoletes and provides as there is no longer a naming conflict. Are both points acceptable for you?

As part of the collaboration with a few projects I'm providing a path to go from the official Nvidia CUDA repositories to my packages, and the official Nvidia repositories use almost (but not entirely) the same naming convention as in RPMFusion. So the need for Obsoletes, Provides and Epoch in case of updates on Nvidia side. Same thing for the CUDA packages. I need to check if this is still needed, it was in 2014. I will revert to you in January, I have no access to my systems before then.

**Comment by scaronni** [comment]
Hi Leigh,

I guess we can remove the Obsoletes/Provides and just leave the Conflicts for everything (glvnd, Nvidia packages, etc.). We can move this discussion off list.
